### PR TITLE
Hide build system internals with `@_spi`

### PR DESCRIPTION
**Pull request by MaxDesiatov**
These details should not leak outside of the SwiftPM package. In the (hopefully temporary) absence of `package` access control, we can hide build system-related APIs with `@_spi` annotations.

### CODE REVIEW

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
@swift-ci test macos

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

File: Tests/CommandsTests/BuildCommandTests.swift [hunk 1]
```diff
@@ -12,7 +12,11 @@
 
 import Basics
 @testable import Commands
-@testable import CoreCommands
+
+@_spi(SwiftPMInternal)
+@testable
```
Comment (bnbarham) [inline Tests/CommandsTests/BuildCommandTests.swift:17]: Would be nice to remove all our @testable too IMO (to avoid the rebuild) and use SPI instead.

**Review by bnbarham — APPROVED** [review]

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Pull request merged**